components/src/preact/aggregatedData/aggregate-bar-chart.tsx

```diff
@@ -0,0 +1,166 @@
+import { BarController, Chart, type ChartConfiguration, type ChartDataset, registerables } from 'chart.js';
+import { type FunctionComponent } from 'preact';
+import { useMemo } from 'preact/hooks';
+
+import type { AggregateData } from '../../query/queryAggregateData';
+import GsChart from '../components/chart';
+import { NoDataDisplay } from '../components/no-data-display';
+import { singleGraphColorRGBAById } from '../shared/charts/colors';
+import { formatProportion } from '../shared/table/formatProportion';
+
+interface AggregateBarChartProps {
+    data: AggregateData;
+    fields: string[];
+    maxNumberOfBars: number;
+}
+
+Chart.register(...registerables, BarController);
+
+type DataPoint = {
+    y: string;
+    x: number;
+    proportion: number;
+};
+
+export const AggregateBarChart: FunctionComponent<AggregateBarChartProps> = ({ data, fields, maxNumberOfBars }) => {
+    if (data.length === 0) {
+        return <NoDataDisplay />;
+    }
+
+    if (fields.length === 0) {
+        return (
+            <NoDataDisplay message='Cannot display a bar chart when there are no fields given that the data should be stratified by.' />
+        );
+    }
+
+    if (fields.length > 2) {
+        return (
+            <NoDataDisplay
+                message={`Cannot display a bar chart when there are more than two fields given that the data should be stratified by. Got: ${fields.join(', ')}`}
+            />
+        );
+    }
+
+    return <AggregateBarChartInner data={data} fields={fields} maxNumberOfBars={maxNumberOfBars} />;
+};
+
+const AggregateBarChartInner: FunctionComponent<AggregateBarChartProps> = ({ data, fields, maxNumberOfBars }) => {
+    const config = useMemo(
+        (): ChartConfiguration<'bar', DataPoint[]> => ({
+            type: 'bar',
+            data: {
+                datasets: getDatasets(fields, maxNumberOfBars, data),
+            },
+            options: {
+                maintainAspectRatio: false,
+                animation: false,
+                indexAxis: 'y',
+                scales: {
+                    x: {
+                        stacked: true,
+                    },
+                    y: {
+                        stacked: true,
+                    },
+                },
+                plugins: {
+                    legend: {
+                        display: false,
+                    },
+                    tooltip: {
+                        mode: 'y',
+                        callbacks: {
+                            label: (context) => {
+                                const { x, proportion } = context.dataset.data[
+                                    context.dataIndex
+                                ] as unknown as DataPoint;
+                                return fields.length === 1
+                                    ? `${x} (${formatProportion(proportion)}})`
```

Comment:
```suggestion
                                    ? `${x} (${formatProportion(proportion)})`
```
I think there is one closing curly bracket too much.